### PR TITLE
Scale slider

### LINKED ISSUE #45: Scale Maximum Number of Features slider in mapping tool
- Scale it based on the actual maximum of features
- get them dynamically fom the api

### PR DESCRIPTION
**Pull request by mehmetcanay**
I implemented the automatic adjustment of the Angular Material Slider for each modality.

closes #45

### CODE REVIEW

**Review by tiadams — CHANGES_REQUESTED** [review]

File: frontend/src/app/mappings/mappings.component.ts [hunk 1]
```diff
@@ -15,10 +15,17 @@ import { MatSliderModule } from '@angular/material/slider';
   styleUrl: './mappings.component.css',
 })
 export class MappingsComponent implements OnInit, OnDestroy {
-  public dataChunks: any[] = [];
-  public maxFeatures: number = 50;
-  public modalities: string[] = [];
-  public noData: boolean = false;
+  dataChunks: any[] = [];
+  // Maximum amount of features to display in a single chords diagram
+  maxFeatures: number = 50;
```
[the remark this answers is not in the document]
Comment (mehmetcanay) [reply frontend/src/app/mappings/mappings.component.ts:20]: This is just for the initialization of the slider. The maximum we can showcase in one diagram is 100. I visually checked it and after 100 it is becoming impossible to interpret.

File: frontend/src/app/mappings/mappings.component.ts [hunk 2]
```diff
@@ -15,10 +15,17 @@ import { MatSliderModule } from '@angular/material/slider';
   styleUrl: './mappings.component.css',
 })
 export class MappingsComponent implements OnInit, OnDestroy {
-  public dataChunks: any[] = [];
-  public maxFeatures: number = 50;
-  public modalities: string[] = [];
-  public noData: boolean = false;
+  dataChunks: any[] = [];
```
[the remark this answers is not in the document]
Comment (mehmetcanay) [reply frontend/src/app/mappings/mappings.component.ts]: Now typed with ChordData interface.

File: frontend/src/app/mappings/mappings.component.ts [hunk 3]
```diff
@@ -15,10 +15,17 @@ import { MatSliderModule } from '@angular/material/slider';
...
+  // Maximum amount of features to display in a single chords diagram
+  maxFeatures: number = 50;
+  // Minimum amount of features for the min variable of the slider
+  minFeatures: number = 20;
+  modalities: string[] = [];
+  modalitiesToCapitalize: string[] = ['apoe', 'csf', 'dti', 'pet'];
```
[the remark this answers is not in the document]
Comment (mehmetcanay) [reply frontend/src/app/mappings/mappings.component.ts:25]: Do you mean in the name of the .csv files or in the endpoint of API?

[the remark this answers is not in the document]
Comment (tiadams) [reply frontend/src/app/mappings/mappings.component.ts:25]: I think the naming of the columns themselves in the CDM, but please discuss this with Yasamin as well

[the remark this answers is not in the document]
Comment (mehmetcanay) [reply frontend/src/app/mappings/mappings.component.ts:25]: The columns are the cohorts. Names of the modalities come from the name of the .csv files, e.g. apoe.csv. Introducing space or uppercase characters here might not be optimal IMO. I will discuss it with @Yasaminsali as well.

**Review by tiadams — APPROVED** [review]

**Pull request merged**